Brightspace/insights-engagement-dashboard
# US121773 [UserDrill] Create the page. Routing via UrlState

[US121773](https://rally1.rallydev.com/#/detail/userstory/449082079828?fdp=true): [Engagement][UserDrill] Create the page

FYI @anhill-D2L @MykolaGalian @rohitvinnakota @hyehayes @bemailloux 

### Functional Testing
* Unit tests
* Manual tests
  * Clicking on user name navigates user to the user view as [per design](https://xd.adobe.com/view/2b998c7a-800e-472b-b41d-d2e14658bff7-7eca/)
  * On dashboard user page, browser's `Back` button navigates back to the default/home view
  * On dashboard user page, clicking on `Back to Engagement Dashboard` navigates to the home view
    * Even if a user navigates to the user page using browser's bookmark or manually typed url
  * The default view popup appears exactly once. It does not appear the second time when a user returns back from the user view
  * On dashboard home page, back button goes to Insights Portal
  * Dashboard page stores its view in URL query parameter. For instance `/ap/visualizations/dashboards/engagement?v=home,0`


### Security Testing - N/A
### Performance Testing - N/A
### Devops - N/A
### UX and docs
- [x] Browsers (Chrome, FF, Edgium, ~Edgacy~)
- [ ] Responsive
- [ ] Accessibility (keyboard nav / screen reading: NVDA + Chrome)
      * ability to navigate link with keyboard

~- [ ] Add filters to user view~

![image](https://user-images.githubusercontent.com/9429561/100230481-73a31200-2f2e-11eb-9b5f-c7d143ffe642.png)
